application/helpers/common_helper.php

```diff
@@ -2927,7 +2927,7 @@ function getTokenFieldsAndNames($surveyid, $bOnlyAttributes = false)
     );
 
     $aExtraTokenFields = getAttributeFieldNames($surveyid);
-    $aSavedExtraTokenFields = Survey::model()->findByPk($surveyid)->tokenAttributes;
+    $aSavedExtraTokenFields = Survey::model()->findByPk($surveyid)->tokenAttributes ?? [];
```

Comment:
https://github.com/LimeSurvey/LimeSurvey/blob/1a6503c0b46d83538afeb84768e5ea927dbc680f/application/models/Survey.php#L672 must return an array, if it's not the case : it's an issue. No ?

And i don't see in function how  it can return something other than an array https://github.com/LimeSurvey/LimeSurvey/blob/1a6503c0b46d83538afeb84768e5ea927dbc680f/application/models/Survey.php#L718-L734